packages/svelte/src/compiler/phases/3-transform/client/visitors/shared/fragment.js

```diff
@@ -139,7 +141,12 @@ function is_static_element(node) {
 		}
 
 		if (attribute.value !== true && !is_text_attribute(attribute)) {
-			return false;
+			// if it's not a text attribute but it's an inlinable expression
+			// we will inline it in the template so we can still consider it static
+			return is_inlinable_expression(
```

Comment:
Returning here with true could lead to bugs down the road in case a condition afterwards would return false

Reply:
But the subsequent checks are only for static text attributes right?

Reply:
Uh i guess except the last one which is about the node name itself...let me see what are we doing in that case

Reply:
Very [good catch](https://svelte.dev/playground/hello-world?version=pr-14249#H4sIAAAAAAAAE22PsWrDQAyGX0XVkhacmKzu-SBDIHu69TK4PpkcXHTGkp0W43cPl9B2aDfx6RO__hm5uRBWeKAYE1zTED08kw9K_gUL7EIkwep9Rv3qs5cBFt9Xu77fyERRM_tohP7jbWIlVsEKjbRD6BUuyY-RrGOnbWJR2B2P-zeowWEjQurw1bEpH3bW2PzOGkkhB0ENq_vHqz_2eWsfjeYsLk-mPG-tY-PDtBYduw40aKR6vgcv1ofJlD9L6xgLVPpUrHQYaTktNxRC8sQpAQAA) !

Reply:
Oh but actually i'm stuped because the rest only checks for the name so it can be problematic there too

Reply:
I moved this check last and moved the check for `node.name` outside the for loop since it didn't had to be inside the loop

Reply:
Oh but wait it's not all actually...because i need to loop over every attribute before returning true

Reply:
Ok now it's fixed for real...i wonder if we should add some snapshots for this case to prevent easy regressions

Reply:
Let's ad some snapshot tests, this code is very fragile without.